### PR TITLE
Rewrite example source to make it simpler and clearer.

### LINKED ISSUE #33265: error CS0103: The name 'Host' does not exist in the current context
In .NET Core 3.1 and .NET 5 I get the compile error:

error CS0103: The name 'Host' does not exist in the current context

I had to install the package Microsoft.AspNetCore.App to compile.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: a4eb702a-d75f-f780-aef1-2e858e14258f
* Version Independent ID: 9a183e07-d5fd-1de9-796d-3bc77f07e033
* Content: [Use dependency injection - .NET](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-usage)
* Content Source: [docs/core/extensions/dependency-injection-usage.md](https://github.com/dotnet/docs/blob/main/docs/core/extensions/dependency-injection-usage.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

---
[Associated WorkItem - 58180](https://dev.azure.com/msft-skilling/Content/_workitems/edit/58180)

### LINKED ISSUE #33313: Console DI Example does too many things
(Unconnected feedback - a general problem here is that clicking feedback goes to this page on the same tab, so you can no longer see the page you are sending feedback on. It would be better to open a new tab.)

This example appears to be doing too many things :-

- It shows how to use DI in a console application
- It explains Singleton, Scoped and Transient
- It does other things I don't understand

In addition :-

- It uses I think a substring of a GUID. Better to think of something better ?
- It uses DefaultImplemenation.cs to add OperationId and MUCH better to add to the interface files individually
- OperationLogger.cs would be MUCH better if it did not try to do everything

Whoever wrote this is clearly very competent, but good documentation, and I can remember the early days of Microsoft when documentation was truly terrible, means editing out all the clever stuff, and leaving something so simple that anyone could understand it, like me.

Remember, users looking at documentation will NOT understand the subject they are looking up. That is why they are looking it up. Make it as easy as possible for them to understand the subject.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: a4eb702a-d75f-f780-aef1-2e858e14258f
* Version Independent ID: 9a183e07-d5fd-1de9-796d-3bc77f07e033
* Content: [Use dependency injection - .NET](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-usage)
* Content Source: [docs/core/extensions/dependency-injection-usage.md](https://github.com/dotnet/docs/blob/main/docs/core/extensions/dependency-injection-usage.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

---
[Associated WorkItem - 58190](https://dev.azure.com/msft-skilling/Content/_workitems/edit/58190)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Rewrite the example source to make it simpler and clearer.

Fixes #33313 and fixes #33265 - [preview](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-usage?branch=pr-en-us-33339)

### CODE REVIEW

File: docs/core/extensions/dependency-injection-usage.md [hunk 1]
```diff
@@ -30,55 +30,73 @@ In this tutorial, you learn how to:
 
 Using either the [dotnet new](../tools/dotnet-new.md) command or an IDE new project wizard, create a new .NET console application named **ConsoleDI.Example**. Add the [Microsoft.Extensions.Hosting](https://www.nuget.org/packages/Microsoft.Extensions.Hosting) NuGet package to the project.
 
+Your new console app project file should resemble the following:
+
+:::code language="xml" source="snippets/configuration/console-di/console-di.csproj":::
+
+> [!IMPORTANT]
+> In this example, the [Microsoft.Extensions.Hosting](https://www.nuget.org/packages/Microsoft.Extensions.Hosting) NuGet package is required to build and run the app. Some metapackages might contain the `Microsoft.Extensions.Hosting` package, in those scenarios and explicit package reference isn't required.
```
Comment (gewarren) [inline docs/core/extensions/dependency-injection-usage.md:38]:
```suggestion
> In this example, the [Microsoft.Extensions.Hosting](https://www.nuget.org/packages/Microsoft.Extensions.Hosting) NuGet package is required to build and run the app. Some metapackages might contain the `Microsoft.Extensions.Hosting` package, in which case an explicit package reference isn't required.
```

File: docs/core/extensions/dependency-injection-usage.md [hunk 2]
```diff
@@ -30,55 +30,73 @@ In this tutorial, you learn how to:
...
+*IReportServiceLifetime.cs*
+
+:::code source="snippets/configuration/console-di/IReportServiceLifetime.cs":::
 
-*IOperation.cs*
+The `IReportServiceLifetime` interface defines:
 
-:::code language="csharp" source="snippets/configuration/console-di/IOperation.cs":::
+- A `Guid Id` property that represents the unique identifier of the service.
+- The <xref:Microsoft.Extensions.DependencyInjection.ServiceLifetime> property that represents the service lifetime.
```
Comment (gewarren) [inline docs/core/extensions/dependency-injection-usage.md:51]:
```suggestion
- A <xref:Microsoft.Extensions.DependencyInjection.ServiceLifetime> property that represents the service lifetime.
```

File: docs/core/extensions/dependency-injection-usage.md [hunk 3]
```diff
@@ -30,55 +30,73 @@ In this tutorial, you learn how to:
...
+## Add a service that requires DI
 
-The `OperationLogger` defines a constructor that requires each of the aforementioned marker interfaces, that is, `ITransientOperation`, `IScopedOperation`, and `ISingletonOperation`. The object exposes a single method that allows the consumer to log the operations with a given `scope` parameter. When invoked, the `LogOperations` method logs each operation's unique identifier with the scope string and message.
+Add the following service lifetime reporter class, which acts as a service to the console app:
+
+*ServiceLifetimeReporter.cs*
+
+:::code source="snippets/configuration/console-di/ServiceLifetimeReporter.cs":::
+
+The `ServiceLifetimeReporter` defines a constructor that requires each of the aforementioned marker interfaces, that is, `IExampleTransientService`, `IExampleScopedService`, and `IExampleSingletonService`. The object exposes a single method that allows the consumer to report on the service with a given `lifetimeDetails` parameter. When invoked, the `ReportServiceLifetimeDetails` method logs each service's unique identifier with the service lifetime message.
```
Comment (gewarren) [inline docs/core/extensions/dependency-injection-usage.md:93]: It looks like you removed the line (65) where marker interfaces were previously mentioned, so I think some further explanation is needed.

File: docs/core/extensions/dependency-injection-usage.md [hunk 4]
```diff
@@ -94,31 +112,17 @@ The app:
...
-
-Scope 2-Call 1 .GetRequiredService<OperationLogger>(): ITransientOperation [ f9af...Always different        ]
-Scope 2-Call 1 .GetRequiredService<OperationLogger>(): IScopedOperation    [ 2bd0...Changes only with scope ]
-Scope 2-Call 1 .GetRequiredService<OperationLogger>(): ISingletonOperation [ 1586...Always the same         ]
-...
-Scope 2-Call 2 .GetRequiredService<OperationLogger>(): ITransientOperation [ fa65...Always different        ]
-Scope 2-Call 2 .GetRequiredService<OperationLogger>(): IScopedOperation    [ 2bd0...Changes only with scope ]
-Scope 2-Call 2 .GetRequiredService<OperationLogger>(): ISingletonOperation [ 1586...Always the same         ]
-```
+In this sample app, you created several interfaces and corresponding implementations. Each of these services is uniquely identified and paired with a <xref:Microsoft.Extensions.DependencyInjection.ServiceLifetime>. The sample app demonstrates registering service implementations against an interface, and how to register pure classes without backing interfaces. The sample app then demonstrates how dependencies defined as constructor parameters are resolved during runtime.
```
Comment (gewarren) [inline docs/core/extensions/dependency-injection-usage.md:115]:
```suggestion
In this sample app, you created several interfaces and corresponding implementations. Each of these services is uniquely identified and paired with a <xref:Microsoft.Extensions.DependencyInjection.ServiceLifetime>. The sample app demonstrates registering service implementations against an interface, and how to register pure classes without backing interfaces. The sample app then demonstrates how dependencies defined as constructor parameters are resolved at run time.
```

File: docs/core/extensions/dependency-injection-usage.md [hunk 5]
```diff
@@ -94,31 +112,17 @@ The app:
...
+
+When the app is ran, it will display output similar to the following:
```
Comment (gewarren) [inline docs/core/extensions/dependency-injection-usage.md:117]:
```suggestion
When you run the app, it displays output similar to the following:
```

**Review by gewarren — APPROVED** [review]

File: docs/core/extensions/dependency-injection-usage.md (hunk 3 above)
Comment (IEvangelist) [inline docs/core/extensions/dependency-injection-usage.md:93]:
```suggestion
The `ServiceLifetimeReporter` defines a constructor that requires each of the aforementioned service interfaces, that is, `IExampleTransientService`, `IExampleScopedService`, and `IExampleSingletonService`. The object exposes a single method that allows the consumer to report on the service with a given `lifetimeDetails` parameter. When invoked, the `ReportServiceLifetimeDetails` method logs each service's unique identifier with the service lifetime message. The log messages help to visualize the service lifetime.
```

**Pull request merged**